Automattic/pocket-casts-android
# Revert SafeDate change because it does not avoid crash

## Description
This reverts https://github.com/Automattic/pocket-casts-android/pull/1573 because that did not end up avoiding the crash. Since it adds a bit of code complexity without bringing any benefit, it's best to remove it.

## Testing Instructions
Just smoke-test the app a bit, and make sure you try "The Daily" and "Stuff You Should Know" podcasts.

## Checklist
